test/test_intensify/test_parallel_hyperband.py

```diff
@@ -0,0 +1,419 @@
+                challengers=challengers,
+                incumbent=None, chooser=None, run_history=self.rh,
+                num_workers=2,
+            )
+            run_infos.append(run_info)
+
+            # Regenerate challenger list
+            challengers = [c for c in challengers if c != run_info.config]
+
+            # Add the config to self.rh in order to make PHB aware that this
+            # config/instance was launched
+            if intent == RunInfoIntent.WAIT:
+                break
+            self.rh.add(
+                config=run_info.config,
+                cost=10,
+                time=0.0,
+                status=StatusType.RUNNING,
+                instance_id=run_info.instance,
+                seed=run_info.seed,
+                budget=run_info.budget,
+            )
+
+        # We have not completed an iteration
+        self.assertEqual(self.PHB.intensifier_instances[0].hb_iters, 0)
+
+        # Because n workers is now 2, we expect 2 sh intensifier_instances
+        self.assertEqual(len(self.PHB.intensifier_instances), 2)
+
+        # Each of the intensifier_instances should have s equal to 1
+        # As no iteration has been completed
+        self.assertEqual(self.PHB.intensifier_instances[0].s_max, 1)
+        self.assertEqual(self.PHB.intensifier_instances[0].s, 1)
+        self.assertEqual(self.PHB.intensifier_instances[1].s_max, 1)
+        self.assertEqual(self.PHB.intensifier_instances[1].s, 1)
+
+        # First let us check everything makes sense in HB-SH-0 HB-SH-0
+        self.assertEqual(self.PHB.intensifier_instances[0].sh_intensifier.initial_budget, 2)
+        self.assertEqual(self.PHB.intensifier_instances[0].sh_intensifier.max_budget, 5)
+        self.assertEqual(len(self.PHB.intensifier_instances[0].sh_intensifier.n_configs_in_stage), 2)
+        self.assertEqual(self.PHB.intensifier_instances[1].sh_intensifier.initial_budget, 2)
+        self.assertEqual(self.PHB.intensifier_instances[1].sh_intensifier.max_budget, 5)
+        self.assertEqual(len(self.PHB.intensifier_instances[1].sh_intensifier.n_configs_in_stage), 2)
+
+        # We are running with:
+        # + 4 runs for sh instance 0 ('all_budgets': array([2.5, 5. ]), 'n_configs_in_stage': [2.0, 1.0])
+        #   that is, for SH0 we run in stage==0 int(2.5) intensifier_instances * 2.0 configs
+        # And this times 2 because we have 2 HB intensifier_instances
+        self.assertEqual(i, 8)
+
+        # Adding a new worker is not possible as we already have 2 intensifier_instances
+        # and n_workers==2
+        intent, run_info = self.PHB.get_next_run(
+            challengers=challengers,
+            incumbent=None, chooser=None, run_history=self.rh,
+            num_workers=2,
+        )
+        self.assertEqual(intent, RunInfoIntent.WAIT)
+
+    def test_add_new_instance(self):
+        """Test whether we can add a SH and when we should not"""
+
+        # By default we do not create a SH
```

Comment:
Here as well. Could you please search and replace all occurences of SH in the hyperband code?

Reply:
I updated the relevant comments. Some are meant for the actual SH instance in the hyperband.